data_steward/analytics/cdr_ops/combined.py

```diff
@@ -857,3 +858,44 @@ def verify_dataset_labels(dataset):
     result.append(query)
 results = '\nUNION ALL\n'.join(result)
 execute(client, results)
+
+# ## Verify no participant in the pdr_ehr_dup_report list has EHR data.
+
+unconsented_records_tpl = JINJA_ENV.from_string("""
+SELECT \'{{domain_table}}\' AS domain_table, person_id
+FROM
+    `{{project}}.{{dataset}}.{{domain_table}}` d
+  JOIN
+    `{{project}}.{{dataset}}.{{mapping_domain_table}}` md
+  USING
+    ({{domain_table}}_id)
+  WHERE
+    person_id IN (
+    SELECT
+      person_id
+    FROM
+      `{{project}}.{{sandbox_dataset}}.{{unconsented_lookup}}`)
+    AND src_dataset_id LIKE '%ehr%'
+""")
+
+# +
+success_msg_if_empty = "All PID's with EHR data are found consenting"
+failure_msg_if_empty = "EHR data is found for PIDs who have not consented to contribute EHR data."
+
+totals = []
+for table in MAPPED_CLINICAL_DATA_TABLES:
+    query = unconsented_records_tpl.render(
+        project=PROJECT_ID,
+        dataset=DATASET_ID,
+        domain_table=table,
+        mapping_domain_table=f'_mapping_{table}',
+        sandbox_dataset=f'{DATASET_ID}_sandbox',
+        unconsented_lookup=UNCONSENTED)
+
+    result = execute(client, query)
+    totals.append(result)
```

Comment:
@MichaelSchmidt82 This is much better.  If a failure happens, the error data is readable.  Whether the test passes or fails is easily discernible from the status check.  The one thing I am going to point out is the execution cycle. 

 As written, this will send, execute, and wait for the results for one query on every loop iteration.  This forces you to aggregate the results in a data frame for eventual usage in the `render_message` function.  I mentioned in our slack conversation that you could do this entire check with one SQL query.  It would take a minor tweak of this code to create a single query instead of multiple queries.  Try to make this improvement.